db/change_cache.go

```diff
@@ -436,16 +436,18 @@ func (c *changeCache) DocChanged(event sgbucket.FeedEvent, docType DocumentType)
 	c.db.DbStats.Database().DCPReceivedCount.Add(1)
 
 	// If the doc update wasted any sequences due to conflicts, add empty entries for them:
-	for _, seq := range syncData.UnusedSequences {
-		base.InfofCtx(ctx, base.KeyCache, "Received unused #%d in unused_sequences property for (%q / %q)", seq, base.UD(docID), syncData.CurrentRev)
-		change := &LogEntry{
-			Sequence:       seq,
-			TimeReceived:   timeReceived,
-			CollectionID:   event.CollectionID,
-			UnusedSequence: true,
+	if len(syncData.UnusedSequences) > 0 {
+		for _, seq := range syncData.UnusedSequences {
+			change := &LogEntry{
+				Sequence:       seq,
+				TimeReceived:   timeReceived,
+				CollectionID:   event.CollectionID,
+				UnusedSequence: true,
+			}
+			changedChannels := c.processEntry(ctx, change)
+			changedChannelsCombined = changedChannelsCombined.Update(changedChannels)
 		}
-		changedChannels := c.processEntry(ctx, change)
-		changedChannelsCombined = changedChannelsCombined.Update(changedChannels)
+		base.InfofCtx(ctx, base.KeyCache, "Received unused sequences %d in unused_sequences property for (%q / %q)", syncData.UnusedSequences, base.UD(docID), syncData.CurrentRev)
```

Comment:
I'd reformat this log just to move the list to the end, since it could get somewhat long. I think this also may make sense to put to Debug since any investigation around specific documents or sequences would require that anyway.

```suggestion
		base.DebugfCtx(ctx, base.KeyCache, "Received unused sequences in unused_sequences property for (%q / %q): %v", base.UD(docID), syncData.CurrentRev, syncData.UnusedSequences)
```